cmd/admin.go

```diff
@@ -165,29 +165,40 @@ func runRepoSyncReleases(c *cli.Context) error {
 		return fmt.Errorf("models.SetEngine: %v", err)
 	}
 
-	repos, count, err := models.SearchRepositoryByName(&models.SearchRepoOptions{})
-	if err != nil {
-		log.Fatal(4, "SearchRepositoryByName: %v", err)
-		return nil
-	}
-	log.Trace("Found %d repos", count)
-	for _, repo := range repos {
-		log.Trace("Synchronizing repo %s with path %s", repo.FullName(), repo.RepoPath())
-		gitRepo, err := git.OpenRepository(repo.RepoPath())
+	log.Trace("Synchronizing repository releases (this may take a while)")
+	for page := 1; ; page++ {
+		repos, count, err := models.SearchRepositoryByName(&models.SearchRepoOptions{
+			Page:     page,
+			PageSize: 10,
```

Comment:
Bigger page size would be better I think (50 for ex.)

Reply:
I was also wondering why 10, only took that value from the example code I was reading: PopulateRepoIndexer here: https://github.com/go-gitea/gitea/blob/f5155b99136b6e19ab494a97adfcee1810a3d5e7/models/repo_indexer.go#L82

But on the other hand, why 50 ? The question is: how much RAM does a single repository information take ? What are we loading, beside the single record, from the DB ? Why not 255 ?

Should we be using a constant for this, to avoid having the same discussion in multiple places ?
For example 20 is used in migration 39: https://github.com/go-gitea/gitea/blob/f5155b99136b6e19ab494a97adfcee1810a3d5e7/models/migrations/v39.go#L34

Reply:
With 59672bf6 I've introduced a new RepositoryListDefaultPageSize constant, set to 64 and used from within the new admin command (to be used by other places too, later)